### PR TITLE
Feature/check shopify hmacs

### PR DESCRIPTION
**Pull request by samutamminen**
Authentication for installation process. Basically followed https://help.shopify.com/api/getting-started/authentication/oauth. In the end we have access token (src/routes/authentication.js line 20) to be saved to database.

### CODE REVIEW

**Comment by samutamminen** [comment]
There is some callback hell, which could be fixed with promises. https://stackoverflow.com/questions/22539815/arent-promises-just-callbacks but got it working faster this way

File: src/routes/authentication.js [hunk 1]
```diff
@@ -6,18 +6,44 @@ const winston = require('winston'); // LOGGING
 const redis = require('../helpers/redisHelper');
 const unirest = require('unirest');
```
Comment (reuterm) [inline src/routes/authentication.js]: Any particular reason you chose 'unirest'? I've been using 'axios' as http library and we should agree on one. According to npm statistics 'axios' is vastly more popular than 'unirest', which is usually a good sign

File: src/helpers/hmacChecker.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const crypto = require('crypto');
+
+const formatKey = (str) => {
+  return str.replace('&', '%26').replace('%','%25')
+};
+
+const formatValue = (str) => {
```
Comment (reuterm) [inline src/helpers/hmacChecker.js]: wouldn't it be much easier (and cleaner) to use a library for en-/decoding?

**Review by reuterm — CHANGES_REQUESTED** [review]
https://help.shopify.com/api/supported-libraries

**Comment by samutamminen** [comment]
I replaced custom url creation and hmac checking with this https://github.com/lpinca/shopify-token, works still like a charm and less complex

File: src/routes/authentication.js [hunk 3]
```diff
@@ -38,7 +38,9 @@ module.exports = function(app) {
           res.status(500).send();
           throw new Error();
         } else {
-          const url = shopifyToken.generateAuthUrl(shop, process.env.SHOPIFY_SCOPES, nonce);
+          const shop_name = (shop.endsWith('myshopify.com')) ?
+            shop.substring(0, shop.length - 13) : shop;
```
Comment (reuterm) [inline src/routes/authentication.js]: you could split by "." and take the first element to make it a bit more robust

**Review by reuterm — APPROVED** [review]

**Pull request merged**